tests/project/test_task1.py

```diff
@@ -1,4 +1,103 @@
+from GramatykiGrafoweAGH.project.task1 import make_initial_graph, P2
+from GramatykiGrafoweAGH import Node, Graph, CannotApplyProductionError
+import pytest
+
+
+def make_P2_left_side_graph():
+    G = Graph()
+
+    x0, y0 = 0, 0
+    level = 1
+
+    x1, y1 = -0.5, -0.5
+    x2, y2 = 0.5, -0.5
+    x3, y3 = -0.5, 0.5
+    x4, y4 = 0.5, 0.5
+
+    I = Node(label='I', x=x0, y=y0, level=level + 1)
+    E1 = Node(label='E', x=x1, y=y1, level=level + 1)
+    E2 = Node(label='E', x=x2, y=y2, level=level + 1)
+    E3 = Node(label='E', x=x3, y=y3, level=level + 1)
+    E4 = Node(label='E', x=x4, y=y4, level=level + 1)
+
+    G.add_nodes([I, E1, E2, E3, E4])
+
+    G.add_edges([
+        (I, E1), (I, E2), (I, E3), (I, E4),
+        (E1, E2), (E2, E4), (E4, E3), (E3, E1),
+    ])
+
+    return G
+
+
+def make_P2_right_side_graph():
+    G = Graph()
+
+    level = 1
+
+    x0, y0 = 0, 0
+    x1, y1 = -0.5, -0.5
+    x2, y2 = 0.5, -0.5
+    x3, y3 = -0.5, 0.5
+    x4, y4 = 0.5, 0.5
+    x5, y5 = 0.0, -0.5
+    x6, y6 = -0.5, 0.0
+    x7, y7 = 0.5, 0.0
+    x8, y8 = 0.0, 0.5
+    x9, y9 = 0.0, 0.0
+
+    I = Node(label='i', x=x0, y=y0, level=level + 1)
+    E1 = Node(label='E', x=x1, y=y1, level=level + 1)
+    E2 = Node(label='E', x=x2, y=y2, level=level + 1)
+    E3 = Node(label='E', x=x3, y=y3, level=level + 1)
+    E4 = Node(label='E', x=x4, y=y4, level=level + 1)
+
+    G.add_nodes([I, E1, E2, E3, E4])
+
+    G.add_edges([
+        (I, E1), (I, E2), (I, E3), (I, E4),
+        (E1, E2), (E2, E4), (E4, E3), (E3, E1),
+    ])
+
+    I1 = Node(label='I', x=(x1 + x9) / 2, y=(y1 + y9) / 2, level=level + 2)
+    I2 = Node(label='I', x=(x2 + x9) / 2, y=(y2 + y9) / 2, level=level + 2)
+    I3 = Node(label='I', x=(x3 + x9) / 2, y=(y3 + y9) / 2, level=level + 2)
+    I4 = Node(label='I', x=(x4 + x9) / 2, y=(y4 + y9) / 2, level=level + 2)
+
+    E1 = Node(label='E', x=x1, y=y1, level=level + 2)
+    E2 = Node(label='E', x=x2, y=y2, level=level + 2)
+    E3 = Node(label='E', x=x3, y=y3, level=level + 2)
+    E4 = Node(label='E', x=x4, y=y4, level=level + 2)
+    E5 = Node(label='E', x=x5, y=y5, level=level + 2)
+    E6 = Node(label='E', x=x6, y=y6, level=level + 2)
+    E7 = Node(label='E', x=x7, y=y7, level=level + 2)
+    E8 = Node(label='E', x=x8, y=y8, level=level + 2)
+    E9 = Node(label='E', x=x9, y=y9, level=level + 2)
+
+    G.add_nodes([
+        I1, I2, I3, I4,
+        E1, E2, E3, E4, E5, E6, E7, E8, E9,
+    ])
+
+    G.add_edges([
+        (I, I1), (I, I2), (I, I3), (I, I4),
+        (I1, E1), (I1, E5), (I1, E9), (I1, E6),
+        (I2, E5), (I2, E2), (I2, E7), (I2, E9),
+        (I3, E6), (I3, E9), (I3, E8), (I3, E3),
+        (I4, E9), (I4, E7), (I4, E4), (I4, E8),
+        (E1, E5), (E5, E2),
+        (E1, E6), (E5, E9), (E2, E7),
+        (E6, E9), (E9, E7),
+        (E6, E3), (E9, E8), (E7, E4),
+        (E3, E8), (E8, E4),
+    ])
+
+    return G
+
+
+def assert_production_cannot_be_applied(P, G):
```

Comment:
Could you please move this function to a new file `src/GramatykiGrafoweAGH/testing.py` and only import it here? Looks like this will be used by other teams as well.